=title Groovy Exception handling (try, catch, Exception)
=timestamp 2019-04-21T15:30:01
=indexes try, catch, Exception
=status show
=books groovy
=author szabgab
=archive 1
=comments_disqus_enable 1

=abstract start
=abstract end

<include file="examples/groovy/divide.groovy">

It works well if the division work well, but:

<code>
$ groovy divide.groovy 3 0

Caught: java.lang.ArithmeticException: Division by zero
java.lang.ArithmeticException: Division by zero
	at divide.div(divide.groovy:2)
	at divide.run(divide.groovy:13)
</code>


We can use <hl>try</hl> and <hl>catch</hl> to catch the exception:

<include file="examples/groovy/catch_exception.groovy">

